### PR TITLE
Add some simple routes to the app.

### PR DESCRIPTION
**Pull request by acabunoc**
* Should probably organize this better into multiple files

### CODE REVIEW

**Comment by acabunoc** [comment]
@alicoding R?

File: app.js [hunk 1]
```diff
@@ -1,23 +1,166 @@
...
+  secret: process.env.BADGES_SECRET || 'supersecret'
 };
 
 var client = new Client(apiEndpoint, auth);
 
+function emailFromORCID (orcid) {
+  return orcid + '@orcid.org';
+}
+
+function ORCIDFromEmail (email) {
```
Comment (alicoding) [inline app.js]: Remove this function because you're not using it.

Comment (alicoding) [inline app.js]: nevermind you'll need it but you just never used it.

File: app.js [hunk 2]
```diff
@@ -1,23 +1,166 @@
...
+  var m;
+  if(m = orcidRe.exec(email) !== null){
+    return m[0]
```
Comment (alicoding) [inline app.js]: This should be `m[1]`;

```
var re = /(\d{4}-\d{4}-\d{4}-\d{4})@orcid\.org/;
re.exec('9020-0003-2222-2323@orcid.org')
Array [ "9020-0003-2222-2323@orcid.org", "9020-0003-2222-2323" ]
```

File: app.js [hunk 3]
```diff
@@ -1,23 +1,166 @@
...
+  }
+}
+
+function modEntry(entry, orcid){
+  entry.orcid = orcid;
+  entry.email = undefined;
```
Comment (alicoding) [inline app.js]: `delete entry.email`

File: app.js [hunk 4]
```diff
@@ -1,23 +1,170 @@
...
+  delete entry.email;
+  return true;
+}
+
+function urlFromDOI(doi){
+  return "http://doi.org/" + decodeURI(doi);
+}
+
+function DOIFromURL(url){
+  return encodeURI(url); // TODO: remove prefix doi.org/
```
Comment (alicoding) [inline app.js]:
```
var Url = require('url');
// pathname should be '/10.1371/journal.pbio.1002126' from 'http://dx.doi.org/10.1371/journal.pbio.1002126'
return encodeURI(Url.parse(url).pathname);
```

File: app.js [hunk 5]
```diff
@@ -1,23 +1,170 @@
...
+}
 
 app.get('/', function(request, response) {
   response.send('Welcome to PaperBadger!');
 });
 
+app.get('/badges', function(request, response){
+  client.getAllBadges({system: system}, function(err, badges){
+    if(err) console.error(err);
```
Comment (alicoding) [inline app.js]: Do we want to `return` if we have an error here?

File: app.js [hunk 6]
```diff
@@ -1,23 +1,170 @@
...
+  })
+});
+
+/* Get badges for a user */
+
+// Get all badge instances earned by a user
+app.get('/users/:orcid/badges', function(request, response){
+  var orcid = request.params.orcid;
+  client.getBadgeInstances({system:system}, {email: emailFromORCID(orcid)}, function(err, badges){
+    if(err) console.error(err);
```
Comment (alicoding) [inline app.js]: same here.

File: app.js [hunk 7]
```diff
@@ -1,23 +1,170 @@
...
+    response.send(badges);
+  })
+});
+
+/* Get badges for a user */
+
+// Get all badge instances earned by a user
+app.get('/users/:orcid/badges', function(request, response){
+  var orcid = request.params.orcid;
```
Comment (alicoding) [inline app.js]: you want to check if orcid is not empty here before making the request. Early return and response with an error if that's the case.

File: app.js [hunk 8]
```diff
@@ -1,23 +1,170 @@
...
+      modEntry(entry, orcid);
+    });
+    response.send(badges);
+  })
+})
+
+// Get all badge instances of a certain badge earned by a user
+app.get('/users/:orcid/badges/:badge', function(request, response){
+  // get all badge instances for the user. Is there a more efficient way to do this?
+  var orcid = request.params.orcid;
```
Comment (alicoding) [inline app.js]: pretty much the same here with orcid and error.

**Comment by alicoding** [comment]
r+ looking good here!

**Pull request merged**